x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/job_messages_pane.tsx

```diff
@@ -46,18 +55,87 @@ export const JobMessagesPane: FC<JobMessagesPaneProps> = React.memo(
 
     const refreshMessage = useCallback(fetchMessages, [jobId]);
 
+    // Clear messages for last 24hrs and refresh jobs list
+    const clearMessages = useCallback(async () => {
+      setIsClearing(true);
+      try {
+        await clearJobAuditMessages(jobId);
+        setIsClearing(false);
+        if (typeof refreshJobList === 'function') {
+          refreshJobList();
+        }
+      } catch (e) {
+        setIsClearing(false);
+        toastNotificationService.displayErrorToast(
+          e,
+          i18n.translate('xpack.ml.jobMessages.clearJobAuditMessagesErrorTitle', {
+            defaultMessage: 'Error clearing job message warnings and errors',
+          })
+        );
+      }
+    }, [jobId]);
+
     useEffect(() => {
       fetchMessages();
     }, []);
 
+    const disabled = messages.length > 0 && messages[0].clearable === false;
```

Comment:
Can we add permission checks to disable the buttons for users with 'viewer' permissions only? For example, I see this for the `ml_viewer` user when hitting the Clear button:

![image](https://user-images.githubusercontent.com/7405507/123839082-a1a56600-d904-11eb-8212-a2216cc2ffe7.png)

![image](https://user-images.githubusercontent.com/7405507/123839149-b255dc00-d904-11eb-9ec7-ab5b1b1fe2a3.png)

Reply:
Added permission check - we now only show the button if the user has 'canCreateJob' permissions. 9538177239a8b6b8f5b5b6ae1d9bf579d223fe41

Reply:
Tested and confirm button will not show up if user only has view permission 👍